.github/workflows/docker-unified.yml

```diff
@@ -125,1041 +118,153 @@ jobs:
-          format: "template"
-          template: "@/contrib/sarif.tpl"
-          output: "trivy-results.sarif"
-          severity: "CRITICAL,HIGH"
-          ignore-unfixed: true
-          vuln-type: "os,library"
-      - name: Upload Trivy scan results to GitHub Security tab
-        uses: github/codeql-action/upload-sarif@v3
-        with:
-          sarif_file: "trivy-results.sarif"
-  datahub_actions_slim_build:
-    name: Build and Push DataHub Actions Slim Docker Image
-    runs-on: depot-ubuntu-24.04
-    needs: [setup]
-    outputs:
-      tag: ${{ steps.tag.outputs.tag }}
-      needs_artifact_download: ${{ needs.setup.outputs.actions_change == 'true' && ( needs.setup.outputs.publish != 'true' && needs.setup.outputs.pr-publish != 'true') }}
-    if: ${{ needs.setup.outputs.actions_change == 'true' || needs.setup.outputs.publish == 'true' || needs.setup.outputs.pr-publish == 'true'}}
-    steps:
-      - name: Check out the repo
-        uses: acryldata/sane-checkout-action@v3
-      - name: Build and push Slim image
-        if: ${{ needs.setup.outputs.actions_change == 'true' }}
-        uses: ./.github/actions/docker-custom-build-and-push
-        with:
-          images: |
-            ${{ env.DATAHUB_ACTIONS_IMAGE }}
-          image_tag: ${{ needs.setup.outputs.slim_tag }}
-          flavor: slim
-          username: ${{ secrets.ACRYL_DOCKER_USERNAME }}
-          password: ${{ secrets.ACRYL_DOCKER_PASSWORD }}
-          build-args: |
-            APP_ENV=slim
-            RELEASE_VERSION=${{ needs.setup.outputs.python_release_version }}
-          publish: ${{ needs.setup.outputs.publish == 'true' || needs.setup.outputs.pr-publish == 'true' }}
-          context: .
-          file: ./docker/datahub-actions/Dockerfile
-          platforms: linux/amd64,linux/arm64/v8
-          depot-project: ${{ vars.DEPOT_PROJECT_ID }}
-      - name: Compute Tag
-        id: tag
-        run: echo "tag=${{ needs.setup.outputs.actions_change == 'true' && needs.setup.outputs.unique_slim_tag || 'head-slim' }}" >> "$GITHUB_OUTPUT"
 
-  datahub_actions_slim_scan:
-    name: "[Monitoring] Scan Datahub Actions Slim images for vulnerabilities"
+  scan_images:
+    name: Scan images for vulnerabilities
     runs-on: depot-ubuntu-24.04
-    needs: [setup, smoke_test_lint, datahub_actions_slim_build]
-    if: ${{ needs.setup.outputs.actions_change == 'true' || needs.setup.outputs.publish == 'true' }}
-    permissions:
-      contents: read # for actions/checkout to fetch code
-      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
-      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+    needs: [setup, base_build, cache_strategy]
+    if: ${{ needs.setup.outputs.publish == 'true' || needs.setup.outputs.pr-publish == 'true'  || github.event_name == 'workflow_dispatch' || github.event_name == 'schedule'}}
```

Comment:
this is a lot of conditions - might be useful to refactor it out into a dedicated var in the setup job